hdk/jack_analyzer/syntax.py

```diff
@@ -0,0 +1,355 @@
+from __future__ import annotations
+
+from collections import UserList
+from dataclasses import dataclass
+from typing import TypeAlias
+from xml.dom.minidom import Document, Element
+from enum import Enum
+
+
+def _add_child(element: Element, tag_name: str, value: str | None) -> Element:
+    dom_tree = element.ownerDocument
+    child = dom_tree.createElement(tag_name)
+    child.appendChild(dom_tree.createTextNode(value))
+    element.appendChild(child)
+    return child
+
+
+@dataclass(frozen=True)
+class Parameter:
+    type_: str
+    var_name: str
+
+
+class ParameterList(UserList[Parameter]):
+    def to_xml(self, dom_tree: Document) -> Element:
+        element = dom_tree.createElement("parameterList")
+        for i, parameter in enumerate(self):
+            element_type = (
```

Comment:
Create to_xml function in Parameter class and move code there. Add property is_identifier in Parameter class and use it in to_xml.